### PR TITLE
SASS-10451: Add EMA Supporting Agent FS and Authorisation logic

### PR DESCRIPTION
**Pull request by joenjones07**
### Description
Implemented secondary agent logic in AuthorisedAction. Added additional tests to cover new logic. Added feature switch for enabling/disabling SupportingAgent

JIRA Story: 
[SASS-10451](https://jira.tools.tax.service.gov.uk/browse/SASS-10451)

Associate App-Config-* PR(s):
- https://github.com/hmrc/app-config-base/pull/11384

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: test/config/MockAppConfig.scala [hunk 1]
```diff
@@ -86,5 +88,7 @@ class MockAppConfig extends AppConfig {
 
     override val useGetCalcListIFPlatform: Boolean = true
 
+    override def emaSupportingAgentsEnabled: Boolean = false
```
Comment (basotia) [inline test/config/MockAppConfig.scala:91]: defined 2 times, please remove one of them

Comment (joenjones07) [reply test/config/MockAppConfig.scala:91]: Both `emaSupportingAgentsEnabled` definitions are needed for this file as removing either one will cause an error

**Review by garytate123 — APPROVED** [review]

File: conf/application.conf [hunk 2]
```diff
@@ -99,4 +99,8 @@ mongodb {
 useBusinessDetailsStub = false
 useEncryption = false
 income-tax-submission-stub-url = "https://income-tax-submission-stub.protected.mdtp:443"
-useGetCalcListIFPlatform = true
\ No newline at end of file
+useGetCalcListIFPlatform = true
+
+feature-switch {
+  ema-supporting-agents-enabled = false
```
Comment (MJCallahanPage) [inline conf/application.conf:105]: Should set this to `true` locally

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
One minor tweak

**Review by basotia — APPROVED** [review]

**Pull request merged**